core/trino-main/src/main/java/io/trino/execution/SetPropertiesTask.java

```diff
@@ -0,0 +1,93 @@
+import io.trino.Session;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.Metadata;
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.metadata.TableHandle;
+import io.trino.security.AccessControl;
+import io.trino.spi.connector.ConnectorMaterializedViewDefinition;
+import io.trino.spi.connector.ConnectorViewDefinition;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.Property;
+import io.trino.sql.tree.SetProperties;
+import io.trino.transaction.TransactionManager;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.util.concurrent.Futures.immediateVoidFuture;
+import static io.trino.metadata.MetadataUtil.createQualifiedObjectName;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.StandardErrorCode.TABLE_NOT_FOUND;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+
+public class SetPropertiesTask
+        implements DataDefinitionTask<SetProperties>
+{
+    @Override
+    public String getName()
+    {
+        return "SET PROPERTIES";
+    }
+
+    @Override
+    public ListenableFuture<Void> execute(
+            SetProperties statement,
+            TransactionManager transactionManager,
+            Metadata metadata,
+            AccessControl accessControl,
+            QueryStateMachine stateMachine,
+            List<Expression> parameters,
+            WarningCollector warningCollector)
+    {
+        Session session = stateMachine.getSession();
+
+        if (statement.getType() == SetProperties.Type.TABLE) {
+            QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
+            Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+            if (tableHandle.isEmpty()) {
+                throw semanticException(TABLE_NOT_FOUND, statement, "Table does not exist: %s", tableName);
+            }
+
+            Optional<ConnectorViewDefinition> view = metadata.getView(session, tableName);
+            if (view.isPresent()) {
+                throw semanticException(NOT_SUPPORTED, statement, "Cannot set properties to a view in ALTER TABLE");
+            }
+
+            Optional<ConnectorMaterializedViewDefinition> materializedView = metadata.getMaterializedView(session, tableName);
+            if (materializedView.isPresent()) {
+                throw semanticException(NOT_SUPPORTED, statement, "Cannot set properties to a materialized view in ALTER TABLE");
+            }
+
+            ImmutableMap.Builder<String, Object> properties = new ImmutableMap.Builder<>();
+            for (Property property : statement.getProperties()) {
+                properties.put(property.getName().getValue(), property.getValue().toString());
+            }
+
+            accessControl.checkCanSetTableProperties(session.toSecurityContext(), tableName, properties.build());
```

Comment:
Do you mean adding below logic to `CreateTableTask` L246?
```
        if (!finalProperties.isEmpty()) {
            accessControl.checkCanSetTableProperties(session.toSecurityContext(), tableName, finalProperties);
        }
```
https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/execution/CreateTableTask.java#L246

Reply:
Yes. Also in since this backward incompatible change as existing queries may start to fail after the udpate let's have a kill switch for that for CREATE TABLE.

Reply:
Also while being here I think we could also do the same for CREATE SCHEMA. So one cannot bypass this security check by using a hive location property on schema level for example. (separate preparatory commit? Also checking access for CREATE TABLE could be a separate commit too so then in ALTER TABLE SET PROPERTIES you would just use access control methods).

Reply:
Do you have recommendation for kill-switch place? AccessControlManager + AccessControlConfig?

Reply:
FeaturesConfig and usage in CreateTableTask. For AlterTableTask I don't think we need this as this is a new feature so no kill switch is needed.

Reply:
Added `security.check-set-properties-permission.create-table` to [FeaturesConfig](https://github.com/trinodb/trino/pull/9401/files#diff-140403fa7b502a89c2ad8238fd3853972e8dc252b48f9cd8daee3140548f88aa) and updated [CreateTableTask](https://github.com/trinodb/trino/pull/9401/files#diff-4421b73df2001755644657bc024d48f3e99cd92a06ac874424b38ac5b9b2cfe3).

Reply:
I think it should be called `legacy.disable-table-properties-security-check-for-create-table`. My point is the `legacy`. This flag should we removed after a year or so, once we learn that it is safe to be removed.

Reply:
How about `deprecated.legacy-disable-table-properties-security-check-for-create-table` considering the consistency with existing configs properties?

Reply:
`deprecated.disable-table-properties-security-check-for-create-table` ? `depreceted.legacy` was a bit too strong to me.

Reply:
Changed to `deprecated.disable-table-properties-security-check-for-create-table`.